src/tooling/grain_doc.md

```diff
@@ -0,0 +1,144 @@
+
+When generating docs, the type of the parameter will be detected by the compiler and inserted into the "Parameters" table.
+
+### @returns
+
+```gr
+/**
+ * @returns description
+ */
+```
+
+The `@returns` annotation indicates the value that is returned by the function that the doc block is documenting.
+
+These must be provided a description of the return value. In the Grain stdlib, we don't use the `@returns` annotation if the function returns `Void`.
+
+When generating docs, the type of the return value will be detected by the compiler and inserted into the "Returns" table.
+
+### @example
+
+```gr
+/**
+ * @example code
+ */
+```
+
+The `@example` annotation can be added to show how the documented code is used. Currently, only single line examples are allowed.
+
+When generating docs, the code snippet will be wrapped in a code block tagged for the `grain` language and inserted into an "Examples" section.
+
+### @since
+
+```gr
+/**
+ * @since version
+ */
+```
+
+The `@since` annotation can be used to specify when the documented code was added.
+
+This must be provided a [Semantic Version](https://semver.org/), optionally prefixed with `v`. For example,`v0.2.0` or `0.2.0` are treated the same.
+
+When generating docs, a `<details>` element titled "Added in X.X.X" will be inserted above the type signature. When using this annotation, you'll be required to specify the `--current-version=X.X.X` flag to the `grain doc` command. This allows you to specify newer versions in your source code, and the generated docs will show "Added in next" instead of the version.
+
+### @history
+
+```gr
+/**
+ * @history version: description
+ */
+```
+
+The `@history` annotation can be used to share significant changes to the documented code.
+
+This must be provided a [Semantic Version](https://semver.org/), optionally prefixed with `v` before the colon (`:`) and a description of the change after the colon.
+
+When generating docs, will add a history table to the `<details>` element created by the `@since` attribute if both were used; otherwise, a new `<details>` element titled  "History" will be inserted above the type signature. When using this attribute, you'll be required to specify the `--current-version=X.X.X` flag to the `grain doc` command. This allows you to specify newer versions in your source code, and the generated docs will show "next" instead of the version.
+
+### @deprecated
+
+```gr
+/***
+ * @deprecated
+ */
+```
+
+The `@deprecated` annotation indicates the deprecation of the documenting export. 
+
+### @module
+
+```gr
+/***
+ * @module Array: Utilities for working with arrays.
+ *
+ * @example import Array from "array"
+ *
+ * @since v0.2.0
+ * @history v0.1.0: Originally named `arrays`
+ * @history v0.2.0: Renamed to `array`
+ */
+```
+
+The `@module` annotation allows you to add header documenting to a module. A module docblock can contain `@example`, `@since`, and `@history` annotations.
```

Comment:
```suggestion
The `@module` attribute allows you to add header documenting to a module. A module docblock can contain `@example`, `@since`, and `@history` attributes.
```